server/src/schema.ts

```diff
@@ -0,0 +1,99 @@
+import type {PGConfig} from './pgconfig/pgconfig.js';
+import type {Executor} from './pg.js';
+
+export async function createDatabase(executor: Executor, dbConfig: PGConfig) {
+  console.log('creating database');
+  const schemaVersion = await dbConfig.getSchemaVersion(executor);
+  if (schemaVersion < 0 || schemaVersion > 1) {
+    throw new Error('Unexpected schema version: ' + schemaVersion);
+  }
+  if (schemaVersion === 0) {
+    await createSchemaVersion1(executor);
+  }
+}
+
+export async function createSchemaVersion1(executor: Executor) {
+  await executor(
+    'CREATE TABLE replicache_meta (key text PRIMARY KEY, value json)',
+  );
+  await executor(
+    "insert into replicache_meta (key, value) values ('schemaVersion', '1')",
+  );
+
+  // cvrversion is null until first pull initializes it.
+  await executor(/*sql*/ `CREATE TABLE replicache_client_group (
+    id VARCHAR(36) PRIMARY KEY NOT NULL,
+    cvrversion INTEGER null,
+    clientversion INTEGER NOT NULL,
```

Comment:
The patch in Replicache contains basically three pieces of data:
1. A patch to the client view
2. A patch to lastMutationIDs
3. A cookie that encodes current state of backend db

todo-row-versioning calculates 1 and 2 using two different systems. For (1) it uses row versioning. For (2) it uses, basically, per-space versioning. Each ReplicaheClientGroup has a version and the clients have the version they updated at.

It would be more elegant to use row-versioning for calculating both patches. This would entail changing the way the version on the ReplicacheClient rows works. Instead of taking the latest value of the RCG.version, it would increment on its own independently, and then be stored as part of the CVR like other data.

The reason I didn't do this originally in todo-row-versioning is that every page load generates a new client. The set of clients only grows, and every CVR would contain the entire set of them. The in-memory diff would eventually get out of control.

But with your implementation here the diff is done inside the database using indexes. I am guessing that it could stay fast indefinitely.

The only problem with my proposal is that with it, the CVR table would grow with O(number-of-clients) rather than as it does now with O(number-of-rows-in-primary-data). But I believe that is addressable. Right now Replicache doesn't completely support deleting old client records (https://github.com/rocicorp/replicache/issues/1033). But if it did, then we could delete them, and they'd automatically get deleted from this CVR table too.